packages/core/src/analytics/integrations/Omnitracking/omnitracking-helper.js

```diff
@@ -321,6 +321,25 @@ export const getValParameterForEvent = (valParameters = {}) => {
   return JSON.stringify(valParameters);
 };
 
+/**
+ * Get the product unit sale price from a specific product.
+ *
+ * @param {object} productData - Product data.
+ *
+ * @returns {*} The unit sale price value.
+ */
+export const getProductUnitSalePrice = productData => {
+  if (typeof productData.unitSalePrice === 'number') {
+    return productData.unitSalePrice;
+  }
+
+  if (productData.priceWithoutDiscount && productData.discountValue) {
```

Comment:
I know this is not possible to happen but if either priceWithoutDiscount or discountValue are 0, the function will return undefined. I think that to be on the safe side we should just check if both are number.

Reply:
Fixed 👍🏼